code/modules/vore/eating/belly_obj_vr.dm

```diff
@@ -97,56 +99,69 @@
 	//List has indexes that are the digestion mode strings, and keys that are lists of strings.
 	var/tmp/list/emote_lists = list()
 
-// Constructor that sets the owning mob
-/datum/belly/New(var/mob/living/owning_mob)
-	owner = owning_mob
-
-// Toggle digestion on/off and notify user of the new setting.
-// If multiple digestion modes are avaliable (i.e. unbirth) then user should be prompted.
-/datum/belly/proc/toggle_digestion()
-	return
-
-// Checks if any mobs are present inside the belly
-// return True if the belly is empty.
-/datum/belly/proc/is_empty()
-	return internal_contents.len == 0
+/obj/belly/initialize()
+	. = ..()
+	//If not, we're probably just in a prefs list or something.
+	if(isliving(loc))
+		owner = loc
+		owner.vore_organs |= src
+		SSbellies.belly_list += src
+
+/obj/belly/Destroy()
+	SSbellies.belly_list -= src
+	if(owner)
+		owner.vore_organs -= src
+		owner = null
+	. = ..()
+
+// Called whenever an atom enters this belly
+/obj/belly/Entered(var/atom/movable/thing,var/atom/OldLoc)
+	if(isliving(thing))
+		var/mob/living/M = thing
+		if(desc)
+			to_chat(M, "<span class='notice'><B>[desc]</B></span>")
+		var/taste
+		if(can_taste && (taste = M.get_taste_message(FALSE)))
+			to_chat(owner, "<span class='notice'>[M] tastes of [taste].</span>")
```

Comment:
It looks like this may print the taste message *before* the "x slides into your y" message.

Reply:
Weird that it doesn't consider this comment outdated, but, changed.